cmd/otel-agent/config/agent_config.go

```diff
@@ -83,8 +98,17 @@ func NewConfigComponent(ctx context.Context, uris []string) (config.Component, e
 	// APM & OTel trace configs
 	pkgconfig.Set("apm_config.enabled", true, pkgconfigmodel.SourceLocalConfigProcess)
 	pkgconfig.Set("apm_config.apm_non_local_traffic", true, pkgconfigmodel.SourceLocalConfigProcess)
+
+	// Port cannot be shared with trace-agent, so just use the next one
+	traceAgentDebugPort := pkgconfig.GetInt("apm_config.debug.port")
+	pkgconfig.Set("apm_config.debug.port", traceAgentDebugPort+1, pkgconfigmodel.SourceLocalConfigProcess) // Default: 5012 is for trace-agent
+
+	otlpTracePort := pkgconfig.GetInt(pkgconfigsetup.OTLPTracePort)
+	pkgconfig.Set(pkgconfigsetup.OTLPTracePort, otlpTracePort+1, pkgconfigmodel.SourceLocalConfigProcess) // Default: 5003 is for trace-agent
```

Comment:
Rethinking on this - we actually do not need the gRPC OTLP port in OTel Agent. In OTel Agent DD exporter, traces are sent to trace agent comp via function calls rather than RPCs https://github.com/DataDog/datadog-agent/blob/e1b7c5943023f1cbee8f51bf5cb34ae77b09ae1e/comp/otelcol/otlp/components/exporter/datadogexporter/traces_exporter.go#L53

So we can just leave the OTLP trace port in trace-agent binary, but disable it in trace-agent component in otel-agent binary

Reply:
Agree with above. We don't debug port or OTLP port for trace agent in OTel agent